src/main.rs

```diff
@@ -0,0 +1,182 @@
+    let _ = transaction.query().insert(data.as_str());
+    transaction.commit().await.unwrap();
+    println!("\nData Loaded Successfully\n");
+    Ok(())
+}
+
+async fn load_schema(connection: Connection)->std::io::Result<()>{
+    let schema = fs::read_to_string("./src/schema.tql")?;
+    // query_options(con.clone()).await.unwrap();
+    let databases = DatabaseManager::new(connection.clone());
+    if databases.contains(TEST_DATABASE).await.unwrap()==false {
+        let _ = databases.create(TEST_DATABASE).await;
+        // define schema
+        let session = Session::new(databases.get(TEST_DATABASE).await.unwrap(), Schema).await.unwrap();
+        let transaction = session.transaction(Write).await.unwrap();
+        transaction.query().define(schema.as_str()).await.unwrap();
+        transaction.commit().await.unwrap();
+        session.force_close().unwrap();
+        load_data(connection.clone()).await?;
+        println!("\nSchema Defined Successfully\n");
+    }else {
+        println!("\nSchema Already Defined\n");
+    }
+
+    Ok(())
+}
+
+async fn query1(connection: Connection)->std::io::Result<()>{    
+    let databases = DatabaseManager::new(connection.clone());
+    let session = Session::new(databases.get(TEST_DATABASE).await.unwrap(), Data).await.unwrap();
+    let transaction = session.transaction(Read).await.unwrap();
+    println!("Query chosen : Get names of all coders with rating >= x");
+    println!("Choose rating x");
+    let x: String=read!("{}\n");
+    let query_part_1 ="match $cc isa coder, has handle $p, has rating >= ";
+    let query_part_2 = "; get $p;";
+    let mut full_query = "".to_owned();
+    full_query = full_query + query_part_1 + x.as_str() + query_part_2;
+
+    println!("{}", full_query);
+
+    let mut answer_stream = transaction.query().match_(&full_query.as_str()).unwrap();
+    while let Some(result) = answer_stream.next().await{
+        match result {
+            Ok(concept_map) => {
+                for (_, concept) in concept_map {
+                    if let Concept::Attribute(Attribute { value: Value::String(value), .. }) = concept {
+                        println!("{}",value);
+                    }
+                }
+            }
+            Err(err) => {
+                panic!("An error occurred fetching answers of a Match query: {err}")
+            }
+        }
+    }
+
+    Ok(())
+}
+
+async fn query2(connection: Connection)->std::io::Result<()>{    
+    let databases = DatabaseManager::new(connection.clone());
+    let session = Session::new(databases.get(TEST_DATABASE).await.unwrap(), Data).await.unwrap();
+    let transaction = session.transaction(Read).await.unwrap();
+    println!("Query chosen : Get IDs of problems with a particular tag");
+    println!("Choose tag");
+    let x: String=read!("{}\n");
+    let query_part_1 ="match $tt ($x, $y) isa possesses-tag; $x isa problem, has problem-number $a; $y isa topic, has topic-name \"";
```

Comment:
You can use `format!` to insert the values into strings. An example is here: https://doc.rust-lang.org/std/macro.format.html